src/libstore/local-store.cc

```diff
@@ -130,6 +130,19 @@ void migrateCASchema(SQLite& db, Path schemaPath, AutoCloseFD& lockFd)
             txn.commit();
         }
 
+        if (curCASchema < 3) {
```

Comment:
Not totally related to this PR, but I wonder if you could take inspiration from the Hydra db migration setup instead of bloating this file with potentially loads of migrations.

Specifically, I'd imagine there would be a directory full of `.sql` schemas alongside the `ca-specific-schema.sql` and this function would instead just sequentially run all of the transactions that haven't happened yet, according to the current schema version.

(Maybe this was talked about in a past PR / issue / the RFC; if so, sorry)

Reply:
Indeed, that could be much nicer. I just blindly copied what already existed for the standard schema :D

Reply:
I created a new issue so that others can chime in with their thoughts / feelings (specifically, Eelco and other source stakeholders): https://github.com/NixOS/nix/issues/5482